src/python/WMCore/MicroService/MSPileup/MSPileup.py

```diff
@@ -4,14 +4,125 @@
 Description: MSPileup provides logic behind the pileup WMCore module.
 """
 
+# system modules
+from threading import current_thread
+
+# 3rd party modules
+import cherrypy
+
 # WMCore modules
+from WMCore.REST.Auth import authz_match
 from WMCore.MicroService.MSCore.MSCore import MSCore
+from WMCore.MicroService.DataStructs.DefaultStructs import PILEUP_REPORT
+from WMCore.MicroService.MSPileup.MSPileupData import MSPileupData
+
+
+def mspileupError(doc):
+    """
+    Check MSPileup record for error
+    :return: None or raise cherrypy.HTTPError
+    """
+    if 'error' in doc:
+        msg = doc['message']
+        code = doc['code']
+        msg = f'MSPileupError: {msg}, code: {code}'
+        raise cherrypy.HTTPError(400, msg) from None
 
 
 class MSPileup(MSCore):
     """
     MSPileup provides whole logic behind the pileup WMCore module.
     """
 
-    def __init__(self, msConfig, logger=None, **kwargs):
-        super(MSPileup, self).__init__(msConfig, **kwargs)
+    def __init__(self, msConfig, **kwargs):
+        # TODO: so far we are testing this service w/o Rucio access
+        super(MSPileup, self).__init__(msConfig, skipRucio=True)
+        # TODO: if more generic approach will be required we'll need to switch to
+#         super(MSPileup, self).__init__(msConfig, **kwargs)
+
+        # for authz we should provide relevant section in MS configuration
+        # if no role/group is provided then authz_match used in HTTP APIs
+        # will always return valid state. Therefore, for test purposes we can
+        # use empty role and group, while for production setup we should set thenm up
+        authzDefaults = msConfig.get('authz_defaults', {})
+        self.role = authzDefaults.get('role', [])
+        self.group = authzDefaults.get('group', [])
+        self.mgr = MSPileupData(msConfig)
+
+    def status(self):
+        """
+        Provide MSPileup status API. We should extend it to check DB connection, etc.
+
+        :return: status dictionary
+        """
+        summary = dict(PILEUP_REPORT)
+        summary.update({'thread_id': current_thread().name})
+        return summary
+
+    def get(self, **kwargs):
```

Comment:
If it does not increase complexity of the code down the road, I would really like to avoid these method names in this class. Names like `get`, `update` are present in many python objects and it always adds extra confusion when reading the code.
How about renaming it get --> getPileup? I will make similar comment to the methods below.

Keep in mind that MSManager will have to be updated (maybe other places as well).

Reply:
done